spec.bs

```diff
@@ -491,19 +491,61 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
-          1. If |lastVersion| is null, or if |lastVersion| does not equal |topic|["{{BrowsingTopic/version}}"], then set |topicParameters|["<code>v</code>"] to |topic|["{{BrowsingTopic/version}}"].
-          1. Set |lastVersion| to |topic|["{{BrowsingTopic/version}}"].
+          1. Increment |serializedTopicsInnerListLength| by the length of |topic|["{{BrowsingTopic/version}}"].
+          1. Increment |serializedTopicsInnerListLength| by |paramsSeparatorLength| + |paramKeyLength|.
+          1. Set |topicParameters|["<code>v</code>"] to a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens">Token</a> having words |topic|["{{BrowsingTopic/version}}"].
           1. Associate |topicParameters| with |topicItem|.
-          1. Insert |topicItem| to |headerStructuredFields|.
+          1. Insert |topicItem| to |topicsInnerList|.
+      1. Let |maxNumberOfEpochs| be 3 (i.e. topics are selected from the last 3 epochs).
+      1. Let |topicMaxLength| be number of digits in the maximum [=browsing topics types/topic id=] (e.g. for <a href="https://github.com/patcg-individual-drafts/topics/blob/main/taxonomy_v1.md">Chrome's initial taxonomy</a>, |topicMaxLength| is 3, as the [=browsing topics types/topic id=] has maximum 3 digits).
+      1. Let |versionMaxLength| be the length of the current [=browsing topics types/version=].
+      1. Let |innerListItemsSeparatorLength| be 1 (i.e. structured fields use a single character (' ') to separate inner list items).
+      1. Let |maxPaddingLength| be <code>|maxNumberOfEpochs| * (|topicMaxLength| + |versionMaxLength| + |paramsSeparatorLength| + |paramKeyLength|) + (|maxNumberOfEpochs| - 1) * |innerListItemsSeparatorLength|</code>.
+      1. Let |paddingLength| be the maximum of |maxPaddingLength| - |serializedTopicsInnerListLength| and 0.
+      1. Let |paddedToken| be "P".
+      1. Append |paddingLength| <code>"0"</code> characters to the end of |paddedToken|.
+      1. Let |headerStructuredFields| be an empty Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-dictionaries">Dictionary</a>.
+      1. Set |headerStructuredFields|["<code>t</code>"] to |topicsInnerList|.
+      1. Set |headerStructuredFields|["<code>p</code>"] to a Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-tokens">Token</a> having words |paddedToken|.
       1. [=Set a structured field value=] given ([:Sec-Browsing-Topics:], |headerStructuredFields|) in |request|'s [=request/header list=].
 
-    Note: In Chrome's experimentation phase, it will additionally require a valid <a href="https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/explainer.md">Origin Trial</a> token to exist in |initiatorWindow|'s <a data-cite="!HTML#concept-document-window">associated document</a> for the request to be eligible for topics.
+    <div class="note">
```

Comment:
Possible idea for spec-writing, feel free to disregard: you could try to generate the padding length by first generating an artificial maximum-length header value, then generating a first draft of your real header value, and then subtracting their serialized lengths. It's inefficient, but it's perhaps easier to understand for the reader, and the implementation could do the more efficient thing.

(You could maybe even generate the artificial maximum-length header value ahead of time, in the terminology section or something?)

Reply:
I incorporated your first suggestion. (I kept the generation of maximum-length inline as I feel it'd give more context here.)